x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals_histogram_panel/signals_histogram/index.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  Axis,
+  Chart,
+  getAxisId,
+  getSpecId,
+  HistogramBarSeries,
+  niceTimeFormatByDay,
+  Settings,
+  timeFormatter,
+} from '@elastic/charts';
+import React, { useEffect, useMemo } from 'react';
+import { EuiLoadingContent } from '@elastic/eui';
+import { isEmpty } from 'lodash/fp';
+import { useQuerySignals } from '../../../../../containers/detection_engine/signals/use_query';
+import { Query } from '../../../../../../../../../../src/plugins/data/common/query';
+import { esFilters, esQuery } from '../../../../../../../../../../src/plugins/data/common/es_query';
+import { SignalsAggregation, SignalsTotal } from '../types';
+import { formatSignalsData, getSignalsHistogramQuery } from './helpers';
+import { useTheme } from '../../../../../components/charts/common';
+import { useKibana } from '../../../../../lib/kibana';
+
+interface HistogramSignalsProps {
+  filters?: esFilters.Filter[];
+  from: number;
+  legendPosition?: 'left' | 'right' | 'bottom' | 'top';
+  loadingInitial: boolean;
+  query?: Query;
+  setTotalSignalsCount: React.Dispatch<SignalsTotal>;
+  stackByField: string;
+  to: number;
+  updateDateRange: (min: number, max: number) => void;
+}
+
+export const SignalsHistogram = React.memo<HistogramSignalsProps>(
+  ({
+    to,
+    from,
+    query,
+    filters,
+    legendPosition = 'bottom',
+    loadingInitial,
+    setTotalSignalsCount,
+    stackByField,
+    updateDateRange,
+  }) => {
+    const [isLoadingSignals, signalsData, setQueryString] = useQuerySignals<{}, SignalsAggregation>(
+      JSON.stringify(getSignalsHistogramQuery(stackByField, from, to, []))
+    );
+    const theme = useTheme();
+    const kibana = useKibana();
+
+    const formattedSignalsData = useMemo(() => formatSignalsData(signalsData), [signalsData]);
+
+    useEffect(() => {
+      setTotalSignalsCount(
+        signalsData?.hits.total ?? {
+          value: 0,
+          relation: 'eq',
+        }
+      );
+    }, [signalsData]);
+
+    useEffect(() => {
+      const converted = esQuery.buildEsQuery(
+        undefined,
+        query != null ? [query] : [],
+        filters?.filter(f => f.meta.disabled === false) ?? [],
+        {
+          ...esQuery.getEsQueryConfig(kibana.services.uiSettings),
+          dateFormatTZ: undefined,
+        }
+      );
+
+      try {
+        setQueryString(
+          JSON.stringify(
```

Comment:
I would push this down more into the bottom of the stack and stringify right before you do a fetch.